tests/load/pipeline/test_pipelines.py

```diff
@@ -994,6 +996,19 @@ def table_3(make_data=False):
         is True
     )
 
+    job_client, _ = pipeline._get_destination_clients(schema)
+
+    if isinstance(job_client, WithStagingDataset):
+        with pipeline.sql_client() as client:
```

Comment:
this is OK but you are hardcoding the staging dataset and qualified table name. this will sometimes work sometimes not depending on the destination. look here:
```
with client.sql_client.with_staging_dataset(
        client.should_load_data_to_staging_dataset(client.schema.tables[table_name])  # type: ignore[attr-defined]
    ):
        canonical_name = client.sql_client.make_qualified_table_name(table_name)
    # write row
    with io.BytesIO() as f:
        write_dataset(client, f, [data_types], column_schemas)
        query = f.getvalue().decode()
    expect_load_file(client, file_storage, query, table_name)
    db_row = list(client.sql_client.execute_sql(f"SELECT * FROM {canonical_name}")[0])
```
it switches to staging dataset to create qualified table name. and then does the select. IMO dremio will start working

Reply:
Okay, let's see